fedora-prime-select

```diff
@@ -1,61 +1,192 @@
 #!/usr/bin/env bash
 set -e
 
-# check if root
+# check if root and ask for password if not
 if [[ $EUID -ne 0 ]]; then
-  printf 'Script must be executed as root.\n'
-  exit 1
+  printf 'Script must be executed as root :\n'
+  su -c $0
+#  exit 1
 fi
 
 clean_files() {
   files=(
     '/etc/X11/xinit/xinitrc.d/nvidia'
+    '/etc/X11/xinit/xinitrc.d/intel'
     '/etc/X11/xorg.conf.d/00-avoid-glamor.conf'
     '/etc/X11/xorg.conf.d/99-nvidia.conf'
     '/etc/X11/xorg.conf'
   )
-
-  for file in "${files[@]}"; do
-    if [[ -h $file ]]; then
-      printf "Deleting %s\n" "$file"
-      rm -f "$file"
-    elif [[ -f $file ]]; then
-      printf "Backing up %s as %s\n" "$file" "$file.fedora-prime.bak"
-      cp "$file" "$file.fedora-prime.bak"
-      rm -f "$file"
-    fi
-  done
+# remove unuseful files, but check
+# if they exist first
+	for file in "${files[@]}"; do
+		if [[ -h $file ]]; then
+			printf "Deleting: %s\n" "$file"
+			[ -e $file ]&& rm -f "$file"
+		elif [[ -f $file ]]; then
+			printf "Backing up: %s as %s\n" "$file" "$file.prime.bak"
+			[ -s $file ]&& cp "$file" "$file.prime.bak"
+			[ -s $file ]&& rm -f "$file"
+		fi
+	done
 }
-
 install_dir='/etc/fedora-prime'
-nvidia32_lib_file='/etc/ld.so.conf.d/nvidia-lib.conf'
+nvidia32_lib_file='/etc/ld.so.conf.d/nvidia-lib32.conf'
 nvidia64_lib_file='/etc/ld.so.conf.d/nvidia-lib64.conf'
+conf=$1
+# default fedora nvidia dirs
+nv_drv_32='/usr/lib/nvidia'
+nv_drv_64='/usr/lib64/nvidia'
+nv_xorg_path='/usr/lib64/nvidia/xorg'
+rc_dir='/etc/rc.d'
+# check any custom dirs for nvidia libraries linking
+if [ -s $install_dir/library.conf ]; then
+	. $install_dir/library.conf
+fi
+# bash lines to insert in rc.local for permanent switch
+bash_line="if [ -e $rc_dir/rc.nvidia ]; then
+    $rc_dir/rc.nvidia &
+fi
+exit 0\n"
 
-case $1 in
-  nvidia )
-    clean_files
+#xrandr_source_cmd=$(xrandr --listproviders| grep "Source"| head -n1| sed -n "s/^.*name://p")
+#xrandr_sink_cmd=$(xrandr --listproviders| grep "Sink"| head -n1| sed -n "s/^.*name://p")
+current_display=$(xrandr --current| grep "connected"| grep "[0-9]\{3,4\}[x]"| sed -n "s/\ (.*$//p")
+connected_display=$(xrandr --current| grep -w "connected"| awk '{print $1}')
+xrandr_outputs(){
+	unset scrn
+	unset def 
+	if [[ $conf == nvidia ]]; then
+		xrandr_source='NVIDIA-0'
+		xrandr_sink='modesetting'
+	else
+		xrandr_source="Intel"
+		xrandr_sink='Intel'
+	fi
+	if [[ $conf == nvidia ]]; then
+		printf "xrandr --setprovideroutputsource $xrandr_sink $xrandr_source\n" > $install_dir/xinitrc.$conf
+		printf "xrandr" >> $install_dir/xinitrc.$conf
+	else
+		printf "xrandr" > $install_dir/xinitrc.$conf	
+	fi
```

Comment:
May be not necessary because xinitrc.intel doesn't like to have xrandr source configured when Intel starts.
So :
```
if [[ $conf == nvidia ]]; then
		xrandr_source='NVIDIA-0'
		xrandr_sink='modesetting'
fi
```
is maybe enough.